benchmarks/run.py

```diff
@@ -8,14 +8,83 @@
 # license that can be found in the LICENSE file or at
 # https://opensource.org/licenses/MIT.
 
+import os
+import sys
+
+# Benchmarks is a scripts folder. However, we want to reuse and import
+# scripts within the benchmarks folder and externally. This is done by adding
+# the benchmarks' parent folder to the system path and using fully qualified
+# module names like benchmarks.configs.names when importing.
+sys.path.insert(
+    0, os.path.dirname(os.path.dirname(os.path.expanduser(os.path.realpath(__file__))))
+)
+
+from dataclasses import fields
+
+from benchmarks.configs.names import (
+    NAMES,
+    MontyWorldExperiments,
+    MontyWorldHabitatExperiments,
+    MyExperiments,
+    PretrainingExperiments,
+    YcbExperiments,
+)
+from tbp.monty.frameworks.config_utils.cmd_parser import create_cmd_parser
 from tbp.monty.frameworks.run_env import setup_env
 
 setup_env()
 
-# Load all experiment configurations from local project
-from configs import CONFIGS  # noqa: E402
-
 from tbp.monty.frameworks.run import main  # noqa: E402
 
 if __name__ == "__main__":
-    main(all_configs=CONFIGS)
+    cmd_args = None
+    cmd_parser = create_cmd_parser(experiments=NAMES)
+    cmd_args = cmd_parser.parse_args()
+    experiments = cmd_args.experiments
+
+    if cmd_args.quiet_habitat_logs:
+        os.environ["MAGNUM_LOG"] = "quiet"
+        os.environ["HABITAT_SIM_LOG"] = "quiet"
+
+    monty_world_experiment_names = [
+        field.name for field in fields(MontyWorldExperiments)
+    ]
+    monty_world_habitat_experiment_names = [
+        field.name for field in fields(MontyWorldHabitatExperiments)
+    ]
+    pretraining_experiment_names = [
+        field.name for field in fields(PretrainingExperiments)
+    ]
+    ycb_experiment_names = [field.name for field in fields(YcbExperiments)]
+    my_experiment_names = [field.name for field in fields(MyExperiments)]
+
+    CONFIGS = dict()
+    for experiment in experiments:
```

Comment:
Is it possible to re-work most of this code both here and in `run_parallel.py` as 1-2 reusable functions? To my eye the main difference is just the presence of the for loop. 

This might help to highlight that this config building code exists, and that if you create a new set of configs (other than `MY_EXPERIMENTS`), then this code will need to be changed (but only once, not both in run and run_paralllel).

This separate file could also include some short documentation outlining this config approach ("Import only the required CONFIGS to limit extraneous imports and therefore dependency on various libraries such as Habitat").

Reply:
Commits changed after rebase; the code was coalesced into `benchmarks/configs/load.py`. The note I was referencing is:
```python
def select_config(experiment: str) -> dict:
    """Select and import the configuration group for the given experiment.
    Args:
        experiment: The name of the experiment to import the configuration for.
    Returns:
        The imported configuration group for the given experiment.
    Note:
        We import the configurations selectively to avoid importing uninstalled
        dependencies. For example, if someone does not install a specific simulator,
        then do not import the corresponding configuration group.
    """
```